capdl-loader-app/src/main.c

```diff
@@ -1332,7 +1332,16 @@ static void init_irq(CDL_Model *spec, CDL_IRQ irq_no)
     if (cdl_irq->slots.num == 1) {
         /* This IRQ is bound. */
         CDL_Cap *endpoint_cap = &cdl_irq->slots.slot[0].cap;
-        seL4_CPtr endpoint_cptr = orig_caps(CDL_Cap_ObjID(endpoint_cap));
+        seL4_CPtr endpoint_cptr;
+
+        seL4_Word badge = get_capData(CDL_Cap_Data(endpoint_cap));
+        if (badge) {
+            endpoint_cptr = (seL4_CPtr)get_free_slot();
+            mint_cap(CDL_Cap_ObjID(endpoint_cap), endpoint_cptr, badge);
```

Comment:
I played around with a change like this a few years ago and decided it was not the right way forward.

By badging the capability at this point, you're taking the badge intended for the copy of the capability in the cspace of the process that wants to receive notifications from this IRQ and applying it to the IRQ copy of the notification. This isn't what the capdl spec actually states.

We ended up working around this limitation in user-level, but discussed adding a new cnode to the capdl spec which represents the kernel's IRQ CNode, and specifying badges in there. 

Let me know if this doesn't make sense and I can try explaining further. It's good to see contributions to the loader!